### PR TITLE
[New Feature] Add operation policy support backend implementation

### PR DESCRIPTION
**Pull request by 1akshitha**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 1]
```diff
@@ -1839,4 +1838,126 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
      * @return List of environments related to the given tenant
      */
     Environment getEnvironment(String organization, String uuid) throws APIManagementException;
+
+    /**
+     * Set existing operation policy mapping to the URI Templates
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:1843]: Shall we make the "T" simple in templates to make it consistent?

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 2]
```diff
@@ -1839,4 +1838,126 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
...
+     *                                  is bit bulky
+     * @return Operation Policy
+     * @throws APIManagementException
+     */
+    OperationPolicyDataHolder getAPISpecificPolicyByPolicyId(String apiUUID, String policyId, String tenantDomain,
+                                                             boolean isWithPolicyDefinition)
+            throws APIManagementException;
+
+    /**
+     * Get the lightweight version of the Operation policies. This will not include the policy definition as it is bulky.
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:1891]: Same here. "O" lower case

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 3]
```diff
@@ -1839,4 +1838,126 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
...
+     * @param tenantDomain        tenant domain
+     * @return List of Operation Policies
+     * @throws APIManagementException
+     */
+    List<OperationPolicyDataHolder> getLightWeightAPISpecificOperationPolicies(String apiUUID, String tenantDomain)
+            throws APIManagementException;
+
+
+    /**
+     * Delete an API Specific policy given by the policy ID. This will delete only the API Specific policies that are
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:1904]: Please check the case scenarios here and other places as well

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -2478,6 +2486,119 @@ private void validateAndSetTransports(APIProduct apiProduct) throws APIManagemen
...
+                    if (policyData != null) {
+                        OperationPolicySpecification policySpecification = policyData.getSpecification();
+                        //Validate the policy applied direction
+                        if (validateAppliedPolicyWithSpecification(policySpecification, policy)) {
+                            policy.setPolicyId(policyData.getPolicyId());
+                            validatedPolicies.add(policy);
+                        }
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Operation policy by the name " + policyName + "is not found for API. Checking shared policies.");
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2519]:
```suggestion
                            log.debug("Operation policy by the name " + policyName + " is not found for API. Checking shared policies.");
```

**Comment by 1akshitha** [comment]
Closing this PR and latest changes will be tracked via https://github.com/wso2/carbon-apimgt/pull/11007

**Pull request closed without merging**